### PR TITLE
SVG Export Plugin locking graph when interrupted

### PR DESCRIPTION
**Pull request by capricornunicorn123**
### Prerequisites

- [ ] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

The SVG Export plugin utilizes `GraphVisualAccess` which holds a lock on the graph during the plugin execution. This PR adds a check for the `SVGGraphBuilder` to allow it to release the lock from the `GraphVisualAccess` when the plugin is interrupted.

### Alternate Designs
Alternate 1 - Refactor the `GraphVisualAccess` to accept a `GraphReadMethods` as in a constructor. This would relinquish responsibility from the `GraphVisualAccess` to release the graph but would extend the class beyond its intended use.

Alternate 2 - Refactor the Export to SVG plugin to use `VisualGraphUtilities` instead of `GraphVisualAccess`. These utilities accept `GraphReadMethods` as parameters however would require a significant amount of work to add required methods for the Export to SVG plugin and would duplicate internal functionality of `GraphVisualAccess`.

### Why Should This Be In Core?
Improves a core feature.

### Benefits
Makes the plugin more usable.

### Possible Drawbacks

May be indicating a larger issue in that `GraphVisualAccess` is not the correct class to use to get graph visual Information by a plugin. No better solution appears to exist.

### Verification Process

1. Open a sphere graph with 10000 nodes
2. Export to SVG
3. Cancel the plugin during execution

### Applicable Issues
#1951

### CODE REVIEW

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/svg/SVGGraphBuilder.java [hunk 1]
```diff
@@ -217,22 +217,28 @@ public SVGGraphBuilder fromPerspective(final AxisConstants exportPerspective) {
...
+
+            // Build the SVG image
+            buildHeader(svgGraph);
+            buildNodes(svgGraph);
+            buildConnections(svgGraph);            
+            buildLayout(svgGraph);
+
+            // Clean up the builder
+            postBuild();
+        } catch (InterruptedException ex) {
```
Comment (antares1470) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/svg/SVGGraphBuilder.java:236]: final

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/svg/SVGGraphBuilder.java [hunk 2]
```diff
@@ -217,22 +217,28 @@ public SVGGraphBuilder fromPerspective(final AxisConstants exportPerspective) {
     public SVGData build() throws InterruptedException {
         
         final SVGObject svgGraph = SVGTemplateConstants.LAYOUT.getSVGObject();
-        
         try{
-            preBuild();
-        } catch (final IllegalArgumentException ex){
-            LOGGER.log(Level.SEVERE, ex.getLocalizedMessage(), ex);
-            return null;
+            try{
```
Comment (antares1470) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/svg/SVGGraphBuilder.java:221]: can you add a space after "try" (I notice a similar case above that may have been missed previously)

**Review by antares1470 — CHANGES_REQUESTED** [review]

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/svg/SVGGraphBuilder.java [hunk 3]
```diff
@@ -217,23 +217,28 @@ public SVGGraphBuilder fromPerspective(final AxisConstants exportPerspective) {
...
+            buildConnections(svgGraph);            
+            buildLayout(svgGraph);
+
+            // Clean up the builder
+            postBuild();     
+        } catch (final InterruptedException ex) {
+            // This class utilizes GraphVisualAccess and is responsible for releasing the graph lock when the plugin is interupted
+            postBuild();
+            throw ex;
         }
```
Comment (OrionsGuardian) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/svg/SVGGraphBuilder.java:240]: Looks like you want the postBuild() to be run at the end in all cases.
Could this be done as a `finally` block at the end a single try statement, without needing to nest an extra try block ?

**Review by OrionsGuardian — CHANGES_REQUESTED** [review]

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/svg/SVGGraphBuilder.java (hunk 3 above)
Comment (capricornunicorn123) [reply CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/svg/SVGGraphBuilder.java:240]: I'll have a play with it and get back to you. As long as the postBuild() in a finally block doesn't cause issues with the catch rethrowing the exception we should be good

**Review by antares1470 — APPROVED** [review]

**Review by OrionsGuardian — APPROVED** [review]

**Pull request merged**